flintrock/core.py

```diff
@@ -504,6 +504,18 @@ def provision_node(
                     sudo sh -c "echo export JAVA_HOME=/usr/lib/jvm/jre >> /etc/environment"
                     source /etc/environment
                 """)
+        print("[{h}] Configuring hostname...".format(h=host))
+        ssh_check_output(
+            client=client,
+            command="""
+                set -e
+
+                fullname=`hostname`.ec2.internal
```

Comment:
`core.py` should not have any logic that's specific to a provider like EC2. If someone added GCE support to Flintrock tomorrow, we'd want to be able to reuse all the logic in `core.py` mostly as-is.